osu.Game/Rulesets/Edit/DistancedHitObjectComposer.cs

```diff
@@ -74,17 +78,78 @@ private void load(OverlayColourProvider colourProvider)
                         Alpha = DistanceSpacingMultiplier.Disabled ? 0 : 1,
                         Child = new EditorToolboxGroup("snapping")
                         {
-                            Child = distanceSpacingSlider = new ExpandableSlider<double, SizeSlider<double>>
+                            Children = new Drawable[]
                             {
-                                Current = { BindTarget = DistanceSpacingMultiplier },
-                                KeyboardStep = adjust_step,
+                                distanceSpacingSlider = new ExpandableSlider<double, SizeSlider<double>>
+                                {
+                                    Current = { BindTarget = DistanceSpacingMultiplier },
+                                    KeyboardStep = adjust_step,
+                                },
+                                currentDistanceSpacingButton = new ExpandableButton
+                                {
+                                    Action = () =>
+                                    {
+                                        (HitObject before, HitObject after)? objects = getObjectsOnEitherSideOfCurrentTime();
+
+                                        Debug.Assert(objects != null);
+
+                                        DistanceSpacingMultiplier.Value = ReadCurrentDistanceSnap(objects.Value.before, objects.Value.after);
+                                        // TODO: This should probably also force distance spacing grid on.
```

Comment:
Would tend to agree with this.

Reply:
I've made this change, but it adds a dependency on #20828 (to centralise the toggle logic).